### PR TITLE
Extend e2e framework to safely give control over SINGULARITY_CACHE_DIR to test developers

### PR DESCRIPTION
**Pull request by gvallee**
**Description of the Pull Request (PR):**

Modifications to the e2e framework to let e2e test developers control the SINGULARITY_CACHE_DIR value when necessary. To achieve this, I extended the use of the test environment that is used to defined a context that is later use to execution one or more Singularity commands.

To give control over the environment variable, a developer simply has to specify the intended value of the environment variable via `env.ImgCacheDir`. That value will be used when starting a Singularity command by safely setting the `SNGULARITY_CACHE_DIR` environment variable.

The PR also includes a test to ensure that the mechanism is properly working: the test create a temporary directory that is used as image cache, we pull a small image and check whether we have the expected entry in the cache.

The same should be done for `SINGULARITY_SYPGPDIR` but will come through a separate PR.

**This fixes or addresses the following GitHub issues:**

- Addresses #4137 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/cmdenvvars/cacheDir.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+			t.Fatalf("failed to delete temporary directory %s: %s", c.env.TestDir, err)
+		}
+	}()
+
+	c.env.ImgCacheDir = cacheDir
+	imgName := "testImg.sif"
+	imgPath := filepath.Join(c.env.TestDir, imgName)
+	cmdArgs := []string{imgPath, "library://alpine:latest"}
+	// Pull an image
+	c.env.RunSingularity(
```
Comment (WestleyK) [inline e2e/cmdenvvars/cacheDir.go:57]: I suggest using https://github.com/sylabs/singularity/blob/master/e2e/internal/e2e/image.go#L41

File: e2e/cmdenvvars/cacheDir.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
+// This software is licensed under a 3-clause BSD license. Please consult the
```
Comment (WestleyK) [inline e2e/cmdenvvars/cacheDir.go:2]: Missing the copyright year thing

```suggestion
// Copyright (c) 2019, Sylabs Inc. All rights reserved.
// This software is licensed under a 3-clause BSD license. Please consult the
```

File: e2e/internal/e2e/singularitycmd.go [hunk 3]
```diff
@@ -20,6 +20,7 @@ import (
 
 	expect "github.com/Netflix/go-expect"
 	"github.com/pkg/errors"
+
```
Comment (WestleyK) [inline e2e/internal/e2e/singularitycmd.go]: Extra newline?

```suggestion
```

**Review by WestleyK — COMMENTED** [review]
Thanks, just a couple suggestions/comments...

File: e2e/internal/e2e/env.go [hunk 4]
```diff
@@ -5,22 +5,16 @@
...
-
+// TestEnv stores all the information under the control of e2e test developers,
+// from specifying which Singularity binary to use to controlling how Singularity
+// environment variables will be set.
 type TestEnv struct {
 	RunDisabled   bool
-	CmdPath       string
-	ImagePath     string
+	CmdPath       string // Path to the Singularity binary to use for the execution of a Singularity command
+	ImagePath     string // Path to the image that has to be used for the execution of a Singularity command
```
Comment (sashayakovtseva) [inline e2e/internal/e2e/env.go:14]: Question: if I test cache by pulling an image, do I need to use `ImagePath`?

**Review by sashayakovtseva — APPROVED** [review]

**Comment by sashayakovtseva** [comment]
Just fix copyright and resolve conflicts, otherwise LGTM.

Need this ASAP to finally merge #4105

**Pull request merged**